// @tidygpt/providers/src/export-provider.ts
import type { ConversationProvider, ProviderHealth, ConversationPage, ConversationFull, ConversationCandidate } from '@tidygpt/shared';

export class ExportProvider implements ConversationProvider {
  id = 'export_provider';
  label = 'Export Analyzer Engine';
  capabilities = {
    listConversations: true,
    readConversation: true,
    archiveConversation: false,
    deleteConversation: false,
    getDates: true,
    getMessageCounts: true,
  };

  private conversations: any[] = [];

  async loadFromJSON(data: any[]) {
    this.conversations = data;
  }

  async healthCheck(): Promise<ProviderHealth> {
    return {
      ok: true,
      version: '1.0',
      source: 'export',
      capabilities: ['listConversations', 'readConversation'],
      warnings: [],
      lastCheckedAt: new Date().toISOString()
    };
  }

  async listConversations(cursor?: string): Promise<ConversationPage> {
    const list = this.conversations.map(c => ({
      id: c.conversation_id || c.id,
      title: c.title || 'Untitled',
      createTime: c.create_time,
      updateTime: c.update_time
    }));
    return { conversations: list };
  }

  async readConversation(id: string): Promise<ConversationFull> {
    const convo = this.conversations.find(c => c.conversation_id === id || c.id === id);
    if (!convo) throw new Error('Not found');
    return {
      id: convo.conversation_id || convo.id,
      title: convo.title || 'Untitled',
      messages: Object.values(convo.mapping || {}).map((m: any) => m.message),
      metadata: {}
    };
  }

  // Returns fully populated candidates
  async generateCandidates(): Promise<ConversationCandidate[]> {
    return this.conversations.map(c => {
      const messages = Object.values(c.mapping || {}).map((m: any) => m.message).filter(Boolean);
      let userMessages = 0;
      let assistantMessages = 0;
      let hasCode = false;

      messages.forEach(m => {
        if (m.author?.role === 'user') userMessages++;
        if (m.author?.role === 'assistant') {
          assistantMessages++;
          // rudimentary code block check in content
          if (m.content?.parts?.some((p: any) => typeof p === 'string' && p.includes('```'))) {
            hasCode = true;
          }
        }
      });

      const candidate: ConversationCandidate = {
        id: c.conversation_id || c.id,
        idHash: c.conversation_id || c.id,
        title: c.title || 'Untitled',
        url: `https://chatgpt.com/c/${c.conversation_id || c.id}`,
        source: "export",
        sourceConfidence: 1.0,
        dates: {
          createdAt: c.create_time ? new Date(c.create_time * 1000).toISOString() : undefined,
          updatedAt: c.update_time ? new Date(c.update_time * 1000).toISOString() : undefined,
          ageDays: c.update_time ? Math.floor((Date.now() / 1000 - c.update_time) / 86400) : undefined,
          dateConfidence: c.update_time ? 1.0 : 0.0,
        },
        counts: {
          userMessages,
          assistantMessages,
          totalMessages: userMessages + assistantMessages,
          countConfidence: 1.0,
        },
        signals: {
          genericTitle: c.title === 'New chat',
          duplicateTitle: false, // Handle in batch later
          hasCode,
          hasFile: "unknown",
          hasImage: "unknown",
          hasArtifact: "unknown",
          isProject: "unknown",
          isCurrentChat: false,
          protectedKeywordMatches: [],
        },
        score: {
          total: 0,
          shortConversation: 0,
          oldAge: 0,
          genericTitle: 0,
          duplicateTitle: 0,
          noFiles: 0,
          noCode: 0,
          noProject: 0,
          noProtectedKeyword: 0,
          lowContentLength: 0,
          confidence: 1.0
        },
        riskFlags: [],
        recommendation: "ignore",
        selectedAction: "none",
        status: "discovered"
      };
      
      return candidate;
    });
  }
}
