// @tidygpt/storage/src/idb-database.ts
import { openDB, DBSchema, IDBPDatabase } from 'idb';
import type { CleanupJob, ConversationCandidate, CleanerSettings, LogEntry } from '@tidygpt/shared';

interface SweeperDB extends DBSchema {
  jobs: {
    key: string;
    value: CleanupJob;
  };
  settings: {
    key: string;
    value: CleanerSettings;
  };
  logs: {
    key: string;
    value: LogEntry;
    indexes: { 'by-job': string };
  };
}

let dbPromise: Promise<IDBPDatabase<SweeperDB>> | null = null;

export async function getDB() {
  if (!dbPromise) {
    dbPromise = openDB<SweeperDB>('ChatSweeperDB', 2, {
      upgrade(db, oldVersion) {
        if (oldVersion < 1) {
          db.createObjectStore('jobs', { keyPath: 'id' });
          db.createObjectStore('settings');
        }
        if (oldVersion < 2) {
          const logsStore = db.createObjectStore('logs', { keyPath: 'id' });
          logsStore.createIndex('by-job', 'jobId');
        }
      },
    });
  }
  return dbPromise;
}

export async function saveJob(job: CleanupJob) {
  const db = await getDB();
  await db.put('jobs', job);
}

export async function getJob(id: string): Promise<CleanupJob | undefined> {
  const db = await getDB();
  return db.get('jobs', id);
}

export async function getAllJobs(): Promise<CleanupJob[]> {
  const db = await getDB();
  return db.getAll('jobs');
}

export async function saveSettings(settings: CleanerSettings) {
  const db = await getDB();
  await db.put('settings', settings, 'user-settings');
}

export async function getSettings(): Promise<CleanerSettings | undefined> {
  const db = await getDB();
  return db.get('settings', 'user-settings');
}

export async function saveLog(log: LogEntry) {
  const db = await getDB();
  await db.put('logs', log);
}

export async function getLogsByJob(jobId: string): Promise<LogEntry[]> {
  const db = await getDB();
  return db.getAllFromIndex('logs', 'by-job', jobId);
}

export async function getAllLogs(): Promise<LogEntry[]> {
  const db = await getDB();
  return db.getAll('logs');
}
